datafusion-examples/examples/simple_udaf.rs

```diff
@@ -108,10 +107,10 @@ impl Accumulator for GeometricMean {
     // This function serializes our state to `ScalarValue`, which DataFusion uses
     // to pass this state between execution stages.
     // Note that this can be arbitrary data.
-    fn state(&self) -> Result<Vec<AggregateState>> {
+    fn state(&self) -> Result<Vec<ScalarValue>> {
```

Comment:
Eventually it might be nice to change this to be `ArrayRef` or something, but we will see what shakes out of https://github.com/apache/arrow-datafusion/issues/2723